fabtests/common/shared.c

```diff
@@ -3069,9 +3120,18 @@ int ft_check_buf(void *buf, int size)
 	static unsigned int iter = 0;
 	int msg_index;
 	int i;
+	int ret;
+
+	if (ft_check_opts(FT_OPT_ENABLE_HMEM)) {
+		ret = ft_hmem_copy_from(opts.iface, opts.device, bounce_buf,
+					buf, size);
+		if (ret != FI_SUCCESS)
+			return ret;
+	}
 
 	msg_index = ((iter++)*INTEG_SEED) % integ_alphabet_length;
-	recv_data = (char *)buf;
+	recv_data = ft_check_opts(FT_OPT_ENABLE_HMEM) ?
+		bounce_buf : (char *) buf;
```

Comment:
We can merge setting recv_data with the if statement above

Reply:
Will fix.